src/uws_client.c

```diff
@@ -1730,26 +1736,15 @@ int uws_client_close_async(UWS_CLIENT_HANDLE uws_client, ON_WS_CLOSE_COMPLETE on
 
             /* Codes_SRS_UWS_CLIENT_01_031: [ `uws_client_close_async` shall close the connection by calling `xio_close` while passing as argument the IO handle created in `uws_client_create`. ]*/
             /* Codes_SRS_UWS_CLIENT_01_368: [ The callback `on_underlying_io_close` shall be passed as argument to `xio_close`. ]*/
-            if (xio_close(uws_client->underlying_io, (on_ws_close_complete == NULL) ? NULL :  on_underlying_io_close_complete, (on_ws_close_complete == NULL) ? NULL : uws_client) != 0)
+            if (xio_close(uws_client->underlying_io, on_underlying_io_close_complete, uws_client) != 0)
```

Comment:
what happen if on_ws_close_complete is null now?

Reply:
These are different callbacks. `on_ws_close_complete` (if nonnull) is called after closing the uws client`, `on_underlying_io_close_complete` is called for the underlying io. I don't know why a not providing an `on_ws_close_complete` should prevent calling the `on_underlying_io_close_complete` callback, which was provided at initialization. The change is necessary since otherwise the `indicate_ws_close_complete` callback, which does the cleanup of the message queue, might not be called, resulting in leaking those messages.